tests/integration/helpers.py

```diff
@@ -72,16 +73,22 @@ async def wait_for_status_message(
         expected_message: The message that vault units should be setting as a status message
         timeout: Wait time, in seconds, before giving up
         cadence: How often to check the status of the units
+        unit_name: The name of the unit to check the status of
 
     Raises:
         TimeoutError: If the expected amount of statuses weren't found in the given timeout.
     """
     seen = 0
     unit_statuses = []
+    if unit_name and count > 1:
+        raise ValueError("Cannot specify unit name and count > 1")
     while timeout > 0:
         unit_statuses = await get_unit_status_messages(ops_test, app_name=app_name)
         seen = 0
-        for unit_name, unit_status_message in unit_statuses:
+        for current_unit_name, unit_status_message in unit_statuses:
+            if unit_name and not count and current_unit_name == unit_name:
```

Comment:
```suggestion
            if unit_name and current_unit_name == unit_name:
```

The check that count was not provided is redundant because it was checked earlier. This will read better.

Reply:
Fixed.